Code/Legacy/CrySystem/LevelSystem/SpawnableLevelSystem.h

```diff
@@ -20,34 +19,40 @@ namespace LegacyLevelSystem
 class SpawnableLevelSystem
         : public ILevelSystem
         , public AzFramework::RootSpawnableNotificationBus::Handler
+        , AzFramework::LevelSystemLifecycleInterface::Registrar
     {
     public:
         explicit SpawnableLevelSystem(ISystem* pSystem);
         ~SpawnableLevelSystem() override;
 
-        // ILevelSystem
+        //! ILevelSystem overrides.
+        //! @{
         void Release() override;
 
         void AddListener(ILevelSystemListener* pListener) override;
         void RemoveListener(ILevelSystemListener* pListener) override;
 
         bool LoadLevel(const char* levelName) override;
```

Comment:
We really just need this level system itself pulled out of legacy, or at least the interface, ILevelSystem, and have it registered as an AZ::Interface. This would be better than having a singleton level interface in the legacy codebase, and another ebus interface in azframework that we combine together in this manner. What we have here, we're going to increasingly need to move methods over to the LevelSystemLifecycle interface to expose it to the rest of the engine, and we'll likely end up needing to rename/refactor the LevelSystemLifecycle interface itself because the added functionality won't be lifecycle specific necessarily, resulting in more interface changes than strictly required.

Reply:
Agreed! My next step was to pull this out entirely and potentially even pull CryCommon out of the other gems that need to use the level system. I’ll create a GHI for that task.

Reply:
https://github.com/o3de/o3de/issues/12772